components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,836 @@
+                errorMsg = " contains the unallowed characters. "
+                return false;
+            }
+            var list = time.split(",");
+            return list.every(function (time){
+                if(time.includes('/')) {
+                    var startingTimeOptionArr = time.split('/');
+                    if(time.endsWith('/')){
+                        errorMsg = " should not end with / "
+                    } else {
+                        errorMsg = " values must lies between 0 and " + maxValue;
+                    }
+                    return validateTimeRange(startingTimeOptionArr, maxValue) ||
+                            (startingTimeOptionArr[0] === '*' && validateTimeRange([startingTimeOptionArr[1]], maxValue));
+                } else if(time.includes('-')) {
+                    var timeRangeArr = time.split('-');
+                    if(time.endsWith('-')){
+                        errorMsg = " should not end with - "
+                    } else {
+                        errorMsg = " values must lies between 0 and " + maxValue;
+                    }
+                    return validateTimeRange(timeRangeArr, maxValue) &&
+                                        (parseInt(timeRangeArr[0])<parseInt(timeRangeArr[1]));
+                } else {
+                    if(time === ""){
+                        errorMsg = " should not end with , "
+                    } else {
+                        errorMsg = " values must lies between 0 and " + maxValue ;
+                    }
+                    return validateTimeRange([time],maxValue) || time === '*';
+                }
+            });
+        };
+
+        var validateTimeRange = function(dataArray, value) {
+            return dataArray.every(element => {
+                return parseInt(element) >= 0  && parseInt(element) <= value;
+            });
+        };
+
+        //this method is used to check the range of dayOfMonth(1-31) ,month(1-12) ,dayOfWeek(1-7)
+        var validateRangeOfDay = function(dataArray, val, endVal) {
+            return dataArray.every(element => {
+                return parseInt(element) >= val && parseInt(element) <= endVal;
+            });
+        };
+
+        var validateDayOfMonthValue = function(dayOfMonth, dayOfWeek) {
+            if(dayOfWeek === '?' && dayOfMonth.includes('L')) {
+                if(dayOfMonth.includes(',') && dayOfWeek === '?'){
+                    return false;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+
+                    return dayOfMonthRangeArr[0] === 'L' && validateRangeOfDay([dayOfMonthRangeArr[1]], 1, 30);
+                }
+            } else if(dayOfMonth.includes('W') && dayOfWeek === '?'){
+                if(dayOfMonth.search(constants.REGEX_FOR_ALPHA_NUMERIC) !== -1 ){
+                    errorMsg = "Invalid Cron Expression";
```

Comment:
In here say "Error: Invalid Cron Expression, Expression cannot contain alpha numeric values " . Try to explain why it is wrong so that user no why. Change all of them

Reply:
fixed in e096cb9